src/DynamoCoreWpf/ViewModels/Search/NodeSearchElementViewModel.cs

```diff
@@ -297,14 +308,11 @@ protected virtual void CreateAndConnectToPort(object parameter)
                 // Create a new node based on node creation name and connection ports
                 dynamoViewModel.ExecuteCommand(new DynamoModel.CreateAndConnectNodeCommand(id, initialNode.GUID,
                       Model.CreationName, 0, portModel.Index, adjustedX, adjustedY, createAsDownStreamNode, false, true));
-            }
-
-            var inputNodes = initialNode.InputNodes.Values.Where(x => x != null).Select(y => y.Item2);
 
-            // Clear current selections and select all input nodes as we need to perform Auto layout on only the input nodes.
-            DynamoSelection.Instance.ClearSelection();
-            DynamoSelection.Instance.Selection.AddRange(inputNodes);
-            DynamoSelection.Instance.Selection.Add(initialNode);
+                //Select all input nodes as we need to perform Auto layout on only the input nodes
+                var inputNodes = initialNode.InputNodes.Values.Where(x => x != null).Select(y => y.Item2);
+                DynamoSelection.Instance.Selection.AddRange(inputNodes);
```

Comment:
I thought these lines should be in else block? Can you test if triggering node autocomplete on out put port would affect input nodes position with these code?

Reply:
Yes I thought so too. I tested it locally and the input nodes position doesn't change. @zeusongit can verify it as well.

Reply:
hmm not sure why that was not going to affect behavior, but I suggest we move the code to `else` block for code clarity?

Reply:
@QilongTang @reddyashish It is already in the else block and only runs for input nodes. `createAsDownStreamNode ` is true when autocomplete is triggered on output ports.

Reply:
Input nodes stay as they are when triggered on outports and vice versa

Reply:
@zeusongit Aaron was saying that lines 312-314 should be inside the else block of createAsDownStreamNode(219-310). I guess, since the nodes are already aligned by auto layout, doing autolayout on the input nodes again wouldn't cause any difference but would be better for the reason Aaron mentioned.

Reply:
@reddyashish @zeusongit Thanks for the clarification!

Reply:
Oh sorry I misunderstood, regarding this, it adds the original parent node on whose port node autocomplete is triggered, so it has to be added in both scenarios and auto layout takes care of the relative position.